src/toil/job.py

```diff
@@ -157,11 +159,65 @@ def _parseResource(name, value):
                             % (name, type(value)))
 
 
-class Job(ResourceRequirementMixin):
+class JobLikeObject(ResourceRequirementMixin):
+    def __init__(self, name, job=None, **requirements):
+        super(JobLikeObject, self).__init__(**requirements)
+        self.name = name
+        self.job = job or self.__class__.__name__
+
+    def __str__(self):
+        printedName = '\''+self.job + '\''
+        if self.name:
+            printedName += ' '+self.name
+        return str(printedName)
+
+class JobNode(JobLikeObject):
+    """
+    This object bridges the job graph, job, and batchsystem classes
+    """
+    def __init__(self, memory, cores, disk, preemptable, job, name, jobStoreID,
+                 command, predecessorNumber=1):
```

Comment:
I think the key to a successful refactoring would be capture all requirements in a dictionary and pass that around, rather than always keeping all requirements as separate kwargs. If that makes sense. I can explain in person tomorrow.